test/test_mag_pruning.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+
+from litgpt.model import GPT, Config
```

Comment:
Shouldn't we use a whittle model?

Reply:
Yes, my original code used a whittle model, so this should be a whittle model

Reply:
ok I changed it whittle model but it does not work,  it says that the gpt model does not have cos and sin attribute and thus wanda and sparse method do not work, this has to do with commit #181  because when I go to the b4 version of the model it works fine. 
if i remove model.cos and model.sin then the function would fail the test because they are required as positional arguments to the layer of the model in the prunning function.

Reply:
I am on it but I have to change some lines GPT block this has to  do with #181

Reply:
I have to call model.reset_prameters()  to initiate model.cos and model.sin for the prunning function to work

Reply:
I think I will be able to proocced when #203 is merged